plugins/tff_backend/models/investor.py

```diff
@@ -13,10 +13,15 @@ class InvestmentAgreement(NdbModel):
     STATUS_SIGNED = 1
     STATUS_PAID = 2
 
+    def _compute_token_count(self):
+        return round(float(self.token_amount) / pow(10, self.token_precision), self.token_precision)
+
     app_user = ndb.UserProperty()
     amount = ndb.FloatProperty(indexed=False)
     token = ndb.StringProperty(indexed=False)
-    token_count = ndb.IntegerProperty(indexed=False)
+    token_count = ndb.ComputedProperty(_compute_token_count, indexed=False)
+    token_amount = ndb.IntegerProperty(indexed=False, default=0)
```

Comment:
I would keep this token_count and add token_count_float or something for the computed

Reply:
I had avoided renaming token_count because that would need a lot of changes everywhere. could rename token_amount to token_count_long though for example

Reply:
What changes? AFAIK the only change thats needed is set token_percision to 0